assets/src/blocks/HubspotForm/HubspotFormEditor.js

```diff
@@ -0,0 +1,147 @@
+import { Fragment, useEffect } from '@wordpress/element';
+import { RichText } from '@wordpress/block-editor';
+import { Sidebar } from './HubspotFormSidebar';
+import { useToAttribute } from './hooks/useToAttribute';
+import { useBackgroundImage } from './hooks/useBackgroundImage';
+import { getStyleFromClassName } from '../getStyleFromClassName';
+
+const { __ } = wp.i18n;
+
+/**
+ * This component is created as a workaround because the <RichText> doesn't provide
+ * The functionality to set max characters like any content editable element.
+ */
+const MaxLengthHelperComponent = ({ value, maxLength, children, darkTheme = false }) => (
+  <div>
+    { children }
+    {maxLength && <span
+      className={`max-length-message ${darkTheme ? 'dark-theme' : ''} ${value.length >= maxLength ? 'max-length-error' : ''}`}>
+      {value.length >= maxLength
+        ? __('Maximum of characters', 'planet4-blocks-backend')
+        : `${value.length} / ${maxLength}`}
+    </span>}
+  </div>
+)
+
+export const HubspotFormEditor = ({
+  attributes: {
+    blockBackgroundImageId,
+    blockBackgroundImageUrl,
+    blockText = '',
+    blockTitle,
+    blockStyle,
+    ctaText,
+    ctaLink,
+    ctaNewTab,
+    formText = '',
+    formTitle,
+    hubspotShortcode,
+    hubspotThankyouMessage,
+    enableCustomHubspotThankyouMessage,
+    className,
+  },
+  setAttributes,
+}) => {
+  const backgroundImage = useBackgroundImage(blockBackgroundImageUrl);
+  const toAttribute = useToAttribute(setAttributes);
+
+  useEffect(() => {
+    if(className) {
+      setAttributes({ blockStyle: getStyleFromClassName(className) });
+    }
+  }, [ className ]);
+
+  return (
+    <Fragment>
+      <Sidebar {...{
+        ctaLink,
+        ctaNewTab,
+        blockBackgroundImageId,
+        enableCustomHubspotThankyouMessage,
+        hubspotThankyouMessage,
+        setAttributes
+      }} />
+      <section className={`hubspot-form hubspot-form-editor block-wide ${blockStyle}`} style={{...backgroundImage}}>
+        <div className='container'>
```

Comment:
You might not need to add the `container` class to the block anymore, since we recently merged changes removing that class from blocks and adding it to the page content element instead 🤔

Reply:
Agree, but if it requires a lot of CSS changes we can also do it in a follow up.

Reply:
Indeed, it's not a blocker!